### DISCUSSION TITLE
rate limit maximum delay time

### DISCUSSION THREAD

**Discussion by JonBendtsen** [drakkan/sftpgo#1136] [Q&A]
Hi

In [Rate limiting](https://github.com/drakkan/sftpgo/blob/main/docs/rate-limiting.md) it says:

`Requests that exceed the configured limit will be delayed or denied if they exceed the maximum delay time.`

but I can not figure out where the maximum delay time is specified or what it is?

[sftpgo.json](https://github.com/drakkan/sftpgo/blob/main/sftpgo.json) only contains the word *delay* once, and that is under `data_provider` so it is not that. *max* is written several times, but none seems to match what I'm looking for. *idle_timeout* is for idle connections, so I would expect that to be already fully connected, and not for connections which are being delayed.

**Comment by drakkan** [comment]
Hello,

the max delay is calculated based on the configured average and period

1 req/s => max delay = 500 ms
100 req/s => max delay = 5 ms

and so on
